### PR TITLE
AD model performance benchmark

### PR DESCRIPTION
**Pull request by kaituo**
### Description
This PR adds a AD model performance benchmark so that we can compare model performance across versions.

For the single stream detector, we refactored tests in DetectionResultEvalutationIT and moved it to SingleStreamModelPerfIT.

For the HCAD detector, we randomly generated synthetic data with known anomalies inserted throughout the signal. In particular, these are one/two/four dimensional data where each dimension is a noisy cosine wave. Anomalies are inserted into one dimension with 0.003 probability. Anomalies across each dimension can be independent or dependent. We have approximately 5000 observations per data set. The data set is generated using the same random seed so the result is comparable across versions.

We also backported https://github.com/opensearch-project/anomaly-detection/pull/600 so that we can capture the performance data in CI output.

Testing done:
1. added unit tests to run the benchmark.

Signed-off-by: Kaituo Li <kaituo@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kaituo** [comment]
Cannot get CI to run due to Common util 1.1.0 build failure. Cannot fix it as it involves changing Common util. Will leave it as it is because 1.1.0 is not actively maintained and verify the build locally.


* What went wrong:
A problem occurred configuring root project 'common-utils'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not find org.opensearch.gradle:build-tools:1.1.0-SNAPSHOT.
     Searched in the following locations:
       - file:/home/runner/.m2/repository/org/opensearch/gradle/build-tools/1.1.0-SNAPSHOT/maven-metadata.xml
       - file:/home/runner/.m2/repository/org/opensearch/gradle/build-tools/1.1.0-SNAPSHOT/build-tools-1.1.0-SNAPSHOT.pom
       - https://repo.maven.apache.org/maven2/org/opensearch/gradle/build-tools/1.1.0-SNAPSHOT/maven-metadata.xml
       - https://repo.maven.apache.org/maven2/org/opensearch/gradle/build-tools/1.1.0-SNAPSHOT/build-tools-1.1.0-SNAPSHOT.pom
       - https://plugins.gradle.org/m2/org/opensearch/gradle/build-tools/1.1.0-SNAPSHOT/maven-metadata.xml
       - https://plugins.gradle.org/m2/org/opensearch/gradle/build-tools/1.1.0-SNAPSHOT/build-tools-1.1.0-SNAPSHOT.pom
       - https://jcenter.bintray.com/org/opensearch/gradle/build-tools/1.1.0-SNAPSHOT/maven-metadata.xml
       - https://jcenter.bintray.com/org/opensearch/gradle/build-tools/1.1.0-SNAPSHOT/build-tools-1.1.0-SNAPSHOT.pom
     Required by:
         project :

**Comment by kaituo** [comment]
Local gradle build:

(22-11-09 14:49:49) <0> [~/code/github/opensearch-ad]
dev-dsk-kaituo-2b-bf84c4db % ./gradlew :build

...
/files-2.1/org.powermock/powermock-reflect/2.0.7/9a8b85397c5a72923962ee9e6bf774e8458803bb/powermock-reflect-2.0.7.jar) to method java.util.Arrays$ArrayList.toArray()
WARNING: Illegal reflective access by org.powermock.reflect.internal.WhiteboxImpl (file:/local/home/kaituo/.gradle/caches/modules-2/files-2.1/org.powermock/powermock-reflect/2.0.7/9a8b85397c5a72923962ee9e6bf774e8458803bb/powermock-reflect-2.0.7.jar) to method java.util.ArrayList.add(java.lang.Object,java.lang.Object[],int)
WARNING: Illegal reflective access by org.powermock.reflect.internal.WhiteboxImpl (file:/local/home/kaituo/.gradle/caches/modules-2/files-2.1/org.powermock/powermock-reflect/2.0.7/9a8b85397c5a72923962ee9e6bf774e8458803bb/powermock-reflect-2.0.7.jar) to method java.util.Arrays$ArrayList.toArray(java.lang.Object[])
WARNING: Illegal reflective access by org.powermock.reflect.internal.WhiteboxImpl (file:/local/home/kaituo/.gradle/caches/modules-2/files-2.1/org.powermock/powermock-reflect/2.0.7/9a8b85397c5a72923962ee9e6bf774e8458803bb/powermock-reflect-2.0.7.jar) to method java.util.Arrays$ArrayList.toArray()
WARNING: Illegal reflective access by org.powermock.reflect.internal.WhiteboxImpl (file:/local/home/kaituo/.gradle/caches/modules-2/files-2.1/org.powermock/powermock-reflect/2.0.7/9a8b85397c5a72923962ee9e6bf774e8458803bb/powermock-reflect-2.0.7.jar) to method java.util.Arrays$ArrayList.toArray(java.lang.Object[])
WARNING: Illegal reflective access by org.powermock.reflect.internal.WhiteboxImpl (file:/local/home/kaituo/.gradle/caches/modules-2/files-2.1/org.powermock/powermock-reflect/2.0.7/9a8b85397c5a72923962ee9e6bf774e8458803bb/powermock-reflect-2.0.7.jar) to method java.util.Arrays$ArrayList.toArray()
WARNING: Illegal reflective access by org.powermock.reflect.internal.WhiteboxImpl (file:/local/home/kaituo/.gradle/caches/modules-2/files-2.1/org.powermock/powermock-reflect/2.0.7/9a8b85397c5a72923962ee9e6bf774e8458803bb/powermock-reflect-2.0.7.jar) to method java.util.ArrayList.add(java.lang.Object,java.lang.Object[],int)
WARNING: Illegal reflective access by org.powermock.reflect.internal.WhiteboxImpl (file:/local/home/kaituo/.gradle/caches/modules-2/files-2.1/org.powermock/powermock-reflect/2.0.7/9a8b85397c5a72923962ee9e6bf774e8458803bb/powermock-reflect-2.0.7.jar) to method java.util.Arrays$ArrayList.toArray(java.lang.Object[])
WARNING: Illegal reflective access by org.powermock.reflect.internal.WhiteboxImpl (file:/local/home/kaituo/.gradle/caches/modules-2/files-2.1/org.powermock/powermock-reflect/2.0.7/9a8b85397c5a72923962ee9e6bf774e8458803bb/powermock-reflect-2.0.7.jar) to method java.util.Arrays$ArrayList.toArray()

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.6.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 11m 48s
28 actionable tasks: 28 executed

**Comment by amitgalitz** [comment]
Just wondering do you know why it is taking so much longer in 1.1 compared to 1.0?

**Review by amitgalitz — APPROVED** [review]

**Review by amitgalitz — APPROVED** [review]

**Comment by kaituo** [comment]
> Just wondering do you know why it is taking so much longer in 1.1 compared to 1.0?

Four changes compared to 1.0:
1)shingle size changed from 1 to 8
2)rcf upgraded from 1.0 to 2.0 where bounding box cache is 0
3)rcf made multiple calls to rcf for explainability and is not as optimized as 3.0

File: build.gradle [hunk 1]
```diff
@@ -69,9 +73,10 @@ tasks.withType(Javadoc) {
 
 repositories {
     mavenLocal()
+    maven { url "https://aws.oss.sonatype.org/content/repositories/snapshots" }
     mavenCentral()
     maven { url "https://plugins.gradle.org/m2/" }
-    jcenter()
+    maven { url "https://d1nvenhzbhpy0q.cloudfront.net/snapshots/lucene/" }
```
Comment (ohltyler) [inline build.gradle:79]: what is the purpose of adding lucene here?

**Review by ohltyler — APPROVED** [review]

**Pull request merged**